prov/gni/src/gnix_datagram.c

```diff
@@ -418,8 +435,94 @@ int gnix_dgram_hndl_free(struct gnix_dgram_hndl *the_hndl)
 	}
 
 err:
-	free(the_hndl->datagram_base);
+	if (ret != FI_SUCCESS)
+		GNIX_INFO(FI_LOG_EP_CTRL, "%s returning error %d\n",
```

Comment:
If I understand what you're trying to do here, you should just need: 
```
GNIX_INFO(FI_LOG_EP_CTRL, "returning error %d\n", ret);
```
Since GNIX_INFO already prints the `__func__ ` variable in the expanded log output.